examples/z_sub_queued.py

```diff
@@ -0,0 +1,97 @@
+#
+# Copyright (c) 2022 ZettaScale Technology
+#
+# This program and the accompanying materials are made available under the
+# terms of the Eclipse Public License 2.0 which is available at
+# http://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+# which is available at https://www.apache.org/licenses/LICENSE-2.0.
+#
+# SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+#
+# Contributors:
+#   ZettaScale Zenoh Team, <zenoh@zettascale.tech>
+#
+
+import sys
+import time
+from datetime import datetime
+import argparse
+import json
+import zenoh
+from threading import Thread
+from zenoh import Reliability, Sample
+
+# --- Command line argument parsing --- --- --- --- --- ---
+parser = argparse.ArgumentParser(
+    prog='z_sub',
+    description='zenoh sub example')
+parser.add_argument('--mode', '-m', dest='mode',
+                    choices=['peer', 'client'],
+                    type=str,
+                    help='The zenoh session mode.')
+parser.add_argument('--connect', '-e', dest='connect',
+                    metavar='ENDPOINT',
+                    action='append',
+                    type=str,
+                    help='Endpoints to connect to.')
+parser.add_argument('--listen', '-l', dest='listen',
+                    metavar='ENDPOINT',
+                    action='append',
+                    type=str,
+                    help='Endpoints to listen on.')
+parser.add_argument('--key', '-k', dest='key',
+                    default='demo/example/**',
+                    type=str,
+                    help='The key expression to subscribe to.')
+parser.add_argument('--config', '-c', dest='config',
+                    metavar='FILE',
+                    type=str,
+                    help='A configuration file.')
+
+args = parser.parse_args()
+conf = zenoh.config_from_file(
+    args.config) if args.config is not None else zenoh.Config()
+if args.mode is not None:
+    conf.insert_json5(zenoh.config.MODE_KEY, json.dumps(args.mode))
+if args.connect is not None:
+    conf.insert_json5(zenoh.config.CONNECT_KEY, json.dumps(args.connect))
+if args.listen is not None:
+    conf.insert_json5(zenoh.config.LISTEN_KEY, json.dumps(args.listen))
+key = args.key
+
+# zenoh-net code  --- --- --- --- --- --- --- --- --- --- ---
```

Comment:
zenoh-net is going to disappearin v0.6.beta1. I'm just leaving the comment as a reminder to do a find and replace in the code once merged.